docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+* All Pet Sets currently require a "governing service", or a Service responsible for the network identity of the pets. The user is responsible for this Service.
+* Updating an existing Pet Set is currently a manual process, meaning you either need to deploy a new Pet Set with the new image version, or orphan Pets one by one, update their image, and join them back to the cluster.
+
+
+## Properties of a Pet
+
+The following examples demonstrate the properties of Pets.
+
+### Identity
+
+A minimal Pet Set might look like:
+
+{% include code.html language="yaml" file="petset.yaml" ghlink="/docs/user-guide/petset.yaml" %}
+
+For completeness it includes the `Service` definition. Lets create the Pet Set:
+
+```shell
+$ kubectl create -f petset.yaml
+service "nginx" created
+petset "nginx" deleted
+```
+
+The identity of a Pet maps to the following:
+
+#### Ordinal index
+you should see 2 pods with predictable names formatted thus: `$(petset name)-$(ordinal index assigned by petset controller)`
+
+```shell
+$ kubectl get po
+NAME      READY     STATUS    RESTARTS   AGE
+web-0     1/1       Running   0          10m
+web-1     1/1       Running   0          10m
+```
+
+#### Stable storage
+2 persistent volumes, one per pod. This is auto created by the Pet Set based on the `volumeTemplate` field
+
+```shell
+$ kubectl get pv
+NAME                                       CAPACITY   ACCESSMODES   STATUS    CLAIM               REASON    AGE
+pvc-90234946-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-0             11m
+pvc-902733c2-3717-11e6-a46e-42010af00002   1Gi        RWO           Bound     default/www-web-1             11m
+```
+
+#### Network identity
+The network identity has 2 parts.
+
+First, we created a headless Service that controls the domain within which we create Pets. The domain managed by this Service takes the form: `$(service name).$(namespace).svc.cluster.local`, where "cluster.local" is the [cluster domain]().
+
+```shell
+$ kubectl get svc
+NAME          CLUSTER-IP     EXTERNAL-IP       PORT(S)   AGE
+nginx         None           <none>            80/TCP    12m
+```
+
+As each pet is created, it gets a matching DNS name carved out of the parent domain, taking the form: `$(petname).$(governing service suffix)`, where the governing service is defined by the `serviceName` field on the Pet Set. Lets verify this assertion with a simple test. The containers are running nginx webservers, which by default will look for an index.html file in `/usr/share/nginx/html/index.html`. That directory is backed by a PersistentVolume created by the Pet Set. So lets write our hostname there (remember the Pet Set gives us a stable hostname):
```

Comment:
Maybe add example here? Table for each value like what the values are for each of these fields (petname, governing service suffix, service name, etc)